### PR TITLE
Add e2e RBD and iSCSI volume tests.

### PR DESCRIPTION
**Pull request by jsafrane**
The test containers needs to run with --privileged, therefore it's disabled by default.

Tested on Fedora 21 and Ubuntu 14.04 as hosts.

### CODE REVIEW

File: contrib/for-tests/volumes-tester/iscsi/Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	# Run as root!
+	./create_block.sh
+
+push: container
+	# Push image tagged as latest to repository
+	gcloud docker push gcr.io/google_containers/volume-iscsi
+	# Push version tagged image to repository (since this image is already pushed it will simply create or update version tag)
+	gcloud docker push gcr.io/google_containers/volume-iscsi:$(TAG)
+
+clean:
```
Comment (brendandburns) [inline contrib/for-tests/volumes-tester/iscsi/Makefile]: need something here?

**Comment by brendandburns** [comment]
LGTM.  I know nothing of Ceph and even less of iSCSI, but good enough to merge for testing purposes.

Do you need someone on the Google side to push the gcr.io/google_containers/... images?

File: contrib/for-tests/volumes-tester/iscsi/Makefile (hunk 1 above)
Comment (jsafrane) [inline contrib/for-tests/volumes-tester/iscsi/Makefile]: Not really, I'm removing it.

**Comment by jsafrane** [comment]
> Do you need someone on the Google side to push the gcr.io/google_containers/... images?

Yes please, @saad-ali did it for my previous NFS and Gluster images.

**Comment by mikedanese** [comment]
@k8s-bot ok to test.

**Comment by mikedanese** [comment]
This is not compiling now for some reason.

**Comment by brendandburns** [comment]
```sh
# github.com/GoogleCloudPlatform/kubernetes/test/e2e
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/volumes.go:369: pod.Status undefined (type string has no field or method Status)
_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/volumes.go:416: pod.Status undefined (type string has no field or method Status)
!!! Error in /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:383
'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${version_ldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
1: /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:383 kube::golang::build_binaries_for_platform(...)
2: /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:521 kube::golang::build_binaries(...)
3: ./hack/build-go.sh:26 main(...)
Exiting with status 1
!!! Error in /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:439
'kube::golang::build_binaries "$@"' exited with status 1
Call stack:
1: /root/workspace/src/github.com/GoogleCloudPlatform/kubernetes/hack/lib/golang.sh:439 kube::golang::build_binaries(...)
2: ./hack/build-go.sh:26 main(...)
Exiting with status 1
```

**Comment by jsafrane** [comment]
It needs a rebase due to PR #11741, I'm working on it.

I got bitten by iSCSI not working through NAT created by the service - it exchanges real IP address of the server pod as part of the protocol.

**Comment by alex-mohr** [comment]
@jsafrane Please let us know when you get a chance to rebase!

**Comment by brendandburns** [comment]
PR changed after LGTM, removing LGTM.

**Comment by brendandburns** [comment]
PR changed after LGTM, removing LGTM.

**Comment by mikedanese** [comment]
@brendanburns I think you broke github api. The PR didn't have LGTM label and you still managed to remove it twice.

**Comment by jsafrane** [comment]
Sorry for the long wait, I had to revert commit 118004c and I wanted to make sure it really works in Vagrant and GCE. A patch that makes this test working on GCE will follow shortly (as a separate PR).

**Comment by brendandburns** [comment]
LGTM.  Shippable is a flake, I kicked it again.

**Comment by jsafrane** [comment]
Thanks for the merge!

@saad-ali, can you please build and push images from `test/images/volumes-tester/rbd` and `test/images/volumes-tester/iscsi` to gcr.io/google_containers?

**Comment by saad-ali** [comment]
@jsafrane Will do

**Comment by saad-ali** [comment]
@jsafrane Both `test/images/volumes-tester/rbd` and `test/images/volumes-tester/iscsi` images are pushed to `gcr.io google_containers`

```
...
5c8faa6d16fa: Image successfully pushed 
Pushing tag for rev [5c8faa6d16fa] on {https://gcr.io/v1/repositories/google_containers/volume-rbd/tags/latest}
Pushing tag for rev [5c8faa6d16fa] on {https://gcr.io/v1/repositories/google_containers/volume-rbd/tags/0.1}
...
93e4a66f9fc5: Image successfully pushed 
Pushing tag for rev [93e4a66f9fc5] on {https://gcr.io/v1/repositories/google_containers/volume-iscsi/tags/latest}
Pushing tag for rev [93e4a66f9fc5] on {https://gcr.io/v1/repositories/google_containers/volume-iscsi/tags/0.1}

```

**Pull request merged**